compiler/rustc_driver/src/lib.rs

```diff
@@ -410,6 +410,10 @@ fn run_compiler(
                 sess.code_stats.print_type_sizes();
             }
 
+            if sess.diagnostic().has_errors_or_lint_errors() {
+                return Err(ErrorReported);
+            }
```

Comment:
You should rarely see an error from codegen, so maybe already stop compilation before codegen?

Reply:
There are still some, so the check still needs to exist after. But I can check more than once, that's not a big deal.

See also https://github.com/rust-lang/rust/pull/87337#issuecomment-884622217, I'm concerned about how this could work for tools.

Reply:
(and it's not that rare - arithmetic_overflow isn't shown until codegen)